crates/ruff_python_parser/src/parser/pattern.rs

```diff
@@ -559,6 +619,40 @@ impl<'src> Parser<'src> {
     ) -> ast::PatternMatchClass {
         let arguments_start = self.node_start();
 
+        let cls = match cls {
+            Pattern::MatchAs(ast::PatternMatchAs {
+                pattern: None,
+                name: Some(ident),
+                ..
+            }) => {
+                if ident.is_valid() {
+                    Box::new(Expr::Name(ast::ExprName {
+                        range: ident.range(),
+                        id: ident.id,
+                        ctx: ExprContext::Load,
+                    }))
+                } else {
+                    Box::new(Expr::Name(ast::ExprName {
+                        range: ident.range(),
+                        id: String::new(),
+                        ctx: ExprContext::Invalid,
+                    }))
+                }
+            }
+            Pattern::MatchValue(ast::PatternMatchValue { value, .. })
+                if matches!(&*value, Expr::Attribute(_)) =>
+            {
+                value
+            }
+            pattern => {
+                self.add_error(
+                    ParseErrorType::OtherError("invalid value for a class pattern".to_string()),
+                    &pattern,
+                );
+                Box::new(recovery::pattern_to_expr(pattern))
+            }
+        };
+
```

Comment:
This is mainly moved from down below because class comes first and then the arguments. It's only to maintain the flow of parsing.

The change here is to use `recovery::pattern_to_expr`.